### PR TITLE
Minor fixes to heatmap post

### PR DESCRIPTION
**Pull request by DayalStrub**
- add requirements.txt
- add explanation for shingles and reference to Manning - refs are cool! and have hover over
- cleaned up text a little.

### CODE REVIEW

**Comment by DayalStrub** [comment]
@gbben did this as a PR into other PR so you can see changes. it's all your good suggestions basically.

File: posts/heatmaps-clustering/index.qmd [hunk 1]
```diff
@@ -4,6 +4,7 @@ subtitle: "A note on clustering distance matrices before plotting. Think sns.clu
 date: "08/03/2022"
 image: "visualization.png"
 categories: [visualisation, how-to]
+bibliography: ../references.bib
```
Comment (DayalStrub) [inline posts/heatmaps-clustering/index.qmd:7]: if we move this to _metadata.yml we wouldn't have to include in each post. metadata does just that - populates all posts with default meta. but not sure whether we will also have some posts where we want them to have their own bib file.

File: posts/heatmaps-clustering/index.qmd [hunk 2]
```diff
@@ -25,12 +25,20 @@ _Question:_ Is there another/better way to re-order the indices? I don't think a
 
 We will use the USA presidential speeches to walk through the clustering, and then do the plotting using altair for interactivity - as much as I like seaborn (and it's [next generation API](https://seaborn.pydata.org/nextgen/) looks very cool), interactivity is great. Then again from the src, it looks like a lot of thought has been put into `clustermap`, so there might be other reasons to use it.
 
-This is therefore a trivial post, but I often find myself wanting to do this. Indeed, I will probably refer back to it myself. The data is chosen arbitrarily, and the reordering doesn't seem to add much in this case, though one could probably have fun checking political party, etc.
+This is therefore a trivial post, but will hopefully turn out to be a useful note for when we want to do this in future. The data is chosen arbitrarily, and the reordering doesn't seem to add much in this case, though one could probably have fun checking political party, etc.
```
Comment (gbben) [inline posts/heatmaps-clustering/index.qmd:28]: Ok so I was looking at the wrong PR 🤦 But good to see you had already modified it. The point about this as a footnote maybe could still be relevant though? Although admittedly it already reads better.

Comment (gbben) [reply posts/heatmaps-clustering/index.qmd:28]: I'd probably still drop the "therefore"

File: posts/references.bib [hunk 3]
```diff
@@ -0,0 +1,6 @@
+@book{schutze2008introduction,
```
Comment (gbben) [inline posts/references.bib:1]: 👏

File: posts/heatmaps-clustering/index.qmd [hunk 4]
```diff
@@ -39,13 +47,14 @@ from scipy.cluster import hierarchy
 
 ## Data
 
-Let us quickly get the presidential addresses from nltk, and then compute the Jaccard distance on the 5-shingles.
+Let us quickly get the presidential addresses from nltk, and then compute the Jaccard distance on the 5-shingles[^1].
+
+[^1]: Note that the data itself is not the focus here. We just need a distance matrix in order to show how it can be reordered. We have chosen to work with a matrix representing the distances between texts in a given corpus, and to consider Jaccard distance on shingles. _Shingles_ are substrings of a certain length that are created by passing a moving window along the text. This choice of distance between texts is common when working with web data, and can deal with mispellings and sequential nature of text (to some extent). See e.g. [@schutze2008introduction, Section 19.6].
```
Comment (gbben) [inline posts/heatmaps-clustering/index.qmd:52]: Eggcellent

File: posts/heatmaps-clustering/index.qmd [hunk 5]
```diff
@@ -39,13 +47,14 @@ from scipy.cluster import hierarchy
...
 
 First we download and import the data.
 
 ```{python}
 #| eval: false
-import nltk
 nltk.download("inaugural")
 ```
```
Comment (gbben) [inline posts/heatmaps-clustering/index.qmd:60]: Can't comments further down but:
1) straight forward -> straightforward
2) there's some commented out code, like the use of pandas `.loc`. Do you want to keep that?
3) you mention "optimal ordering" -- maybe worth a few more words on what you mean?
4) after you plot by data and by distance you could maybe spare a few words for a conclusion/description of the results?

**Review by gbben — APPROVED** [review]
Looks noice! Left a few comments.

**Pull request merged**